### ISSUE TITLE
with-transaction not working as expected

### ISSUE THREAD

**Issue by piczmar** [gorillalabs/neo4j-clj#26]
I have the following function (based on the provided example in `examples` folder)

```
  (db/with-transaction local-db tx
    (create-user tx {:user {:first-name "Luke" :last-name "Skywalker"}})
    (create-user tx {:user {:first-name "John" :last-name "Johnson"}})
    (throw (Exception. "my exception message"))
    (create-user tx {:user {:first-name "Marry" :last-name "Poppins"}}))
```

I execute it on an empty database. I'd expect after the exception is thrown the DB remains empty however it contains the users created before the exception is thrown:

```
({:user {:first-name Luke, :last-name Skywalker}} {:user {:first-name John, :last-name Johnson}})
```
Is it my misunderstanding of `with-transaction` usage, configuration problem or a bug in the library?

**Comment by chrisbetz** [comment]
I will be looking into this - I can confirm the behavior you see, however, I need to find some spare time.

**Comment by chrisbetz** [comment]
Fixed with v5.0.0.
